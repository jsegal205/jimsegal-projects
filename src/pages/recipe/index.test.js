import React from "react";
import { render, cleanup, shallowMount } from "@testing-library/react";
import Recipe from "./index";
import NotFound from "../not-found";

afterEach(cleanup);

describe("Recipe Component", () => {
  describe("when props do not contain `state`", () => {
    it("renders NotFound component", () => {
      const props = {
        location: {}
      };
      const { getByText } = render(<Recipe {...props} />);

      getByText("Not Found");
    });
  });

  describe("when props contain `state`", () => {
    it("renders correctly", () => {
      const props = {
        location: {
          state: {
            title: "title",
            referenceLink: "referenceLink",
            ingredients: "ingredients",
            directions: "directions"
          }
        }
      };
      const { asFragment } = render(<Recipe {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
