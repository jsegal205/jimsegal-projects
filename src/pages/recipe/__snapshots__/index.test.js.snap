// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Component when props contain \`state\` renders correctly 1`] = `
<DocumentFragment>
  <section>
    <h2>
      title
    </h2>
    <small>
      <a
        href="referenceLink"
        rel="noopener noreferrer"
        target="_blank"
      >
        Original Reference
      </a>
    </small>
    <h3>
      Ingredients
    </h3>
    <div
      class="ingredients"
    >
      <p>
        ingredients
      </p>
    </div>
    <h3>
      Directions
    </h3>
    <div
      class="directions"
    >
      <p>
        directions
      </p>
    </div>
  </section>
</DocumentFragment>
`;
